File: Application/src/main/java/com/alphaz/application/authorization/config/OauthClientConfig.java
package com.alphaz.application.authorization.config;

import org.springframework.context.annotation.Bean;
import org.springframework.security.oauth2.client.OAuth2ClientContext;
import org.springframework.security.oauth2.client.OAuth2RestOperations;
import org.springframework.security.oauth2.client.OAuth2RestTemplate;
import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
import org.springframework.security.oauth2.client.token.grant.code.AuthorizationCodeResourceDetails;

import java.util.Arrays;

/**
 * @Author: c0der
 * @Date: 上午11:25 2018/2/6
 * @Description:
 */
//@Configuration
//@EnableOAuth2Client
public class OauthClientConfig {


    @Bean
    public OAuth2RestOperations restTemplate(OAuth2ClientContext oauth2ClientContext) {
        return new OAuth2RestTemplate(resource(), oauth2ClientContext);
    }

    private OAuth2ProtectedResourceDetails resource() {
        AuthorizationCodeResourceDetails resource = new AuthorizationCodeResourceDetails();
        resource.setClientId("");
        resource.setClientSecret("");
        resource.setAccessTokenUri("");
        resource.setUserAuthorizationUri("");
        resource.setScope(Arrays.asList("read"));

        return resource;
    }
}

File: Application/src/main/kotlin/com/alphaz/application/config/OauthClientConfig.kt
package com.alphaz.application.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.oauth2.client.OAuth2ClientContext
import org.springframework.security.oauth2.client.OAuth2RestOperations
import org.springframework.security.oauth2.client.OAuth2RestTemplate
import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails
import org.springframework.security.oauth2.client.token.grant.code.AuthorizationCodeResourceDetails
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client
import java.util.*

/**
 *@Author: c0der
 *@Date: 下午1:25 2018/3/20
 *@Description:
 */
//@Configuration
//@EnableOAuth2Client
open class OauthClientConfig {
    @Bean
    open fun restTemplate(oauth2ClientContext: OAuth2ClientContext): OAuth2RestOperations {
        return OAuth2RestTemplate(resource(), oauth2ClientContext)
    }

    private fun resource(): OAuth2ProtectedResourceDetails {
        val resource = AuthorizationCodeResourceDetails()
        resource.clientId = ""
        resource.clientSecret = ""
        resource.accessTokenUri = ""
        resource.userAuthorizationUri = ""
        resource.scope = Arrays.asList("read")

        return resource
    }
}